coderd/init_script_test.go

```diff
@@ -0,0 +1,31 @@
+package coderd_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/coderdtest"
+)
+
+func TestInitScript(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OK", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, nil)
+		script, err := client.InitScript(context.Background(), "windows", "amd64")
+		require.NoError(t, err)
+		require.NotEmpty(t, script)
```

Comment:
Should test the replacements as well, probably search for substrings like `$env:CODER_AGENT_AUTH_TYPE="token"` (or whatever it's supposed to look like) etc.